### PR TITLE
Implement deletion of annotations and comments

### PR DESCRIPTION
**Pull request by lowjoel**

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq any comments?

File: spec/helpers/course/assessment/answer/programming_helper_spec.rb [hunk 1]
```diff
@@ -17,7 +17,9 @@ def hello:
         end
       end
 
+      before { controller.define_singleton_method(:current_user) { nil } }
```
Comment (allenwq) [inline spec/helpers/course/assessment/answer/programming_helper_spec.rb]: @lowjoel What's this for ?

Comment (lowjoel) [inline spec/helpers/course/assessment/answer/programming_helper_spec.rb]: so the helper can use can?. the helper doesn't have a controller so current_user returns a TypeError.

Comment (allenwq) [inline spec/helpers/course/assessment/answer/programming_helper_spec.rb]: Okay

File: app/views/course/assessment/answer/programming/_annotation_post.html.slim [hunk 2]
```diff
@@ -1,4 +1,9 @@
-= div_for(annotation_post) do
+= div_for(annotation_post, 'data-post-id' => annotation_post.id) do
+  div.toolbar.pull-right
+    - if can?(:destroy, annotation_post)
+      = link_to('#', class: ['delete']) do
```
Comment (allenwq) [inline app/views/course/assessment/answer/programming/_annotation_post.html.slim]: @lowjoel we have two versions of `link_to` in our codebase now (with brackets and without brackets), think it need to standardised.

Comment (lowjoel) [inline app/views/course/assessment/answer/programming/_annotation_post.html.slim]: yeah, hmm.

**Pull request merged**